docs/TrafficSplitting.md

```diff
@@ -0,0 +1,163 @@
+- The order of the flags specified in the command are **not** taken into account.
+- Enlisting the precedence of flag(s) as they are evaluated by `kn`:
+
+1. `--untag-revision` : First, all the referenced tags by this flag are removed from the traffic block
+2. `--tag-revision` or `--tag-latest` : Second, all the revisions are tagged as specified in the traffic block
+3. `--traffic` or `--traffic-latest`: Third, all the referenced revisions (or tags) are assigned given portion of traffic split
+
+Thus, if a user wants to update tag for revision `echo-v2` from `new` to `old` with `50%` traffic while adjusting rest to latest revision, will do
+```
+kn service update svc --untag-revision new --tag-revision echo-v2:old --traffic old:50 --traffic-latest 50
+```
+
+## Summary
+Following table shows quick summary for traffic splitting flags, their value formats, operation to perform and *Repetition* column denotes if repeating the flag is allowed in a `kn service update` command.
+
+
+Flag | Value(s) | Operation | Repetition
+:--- | :---: | :--- | :---:
+`--traffic` | `RevisionName:Percent` | Give `Percent` traffic to `RevisionName` | :heavy_check_mark:
+`--traffic` | `Tag:Percent` | Give `Percent` traffic to the Revision having `Tag` | :heavy_check_mark:
+`--traffic-latest` | `Percent` | Give `Percent` traffic to the latest ready Revision | :heavy_multiplication_x:
+`--tag-revision` | `RevisionName:Tag` | Give `Tag` to `RevisionName` | :heavy_check_mark:
+`--tag-latest` | `Tag` | Give `Tag` to the latest ready Revision | :heavy_multiplication_x:
+`--untag-revision` | `Tag` | Remove `Tag` from Revision | :heavy_check_mark:
+
+
+## Examples
+1. Tag revisions `echo-v1` and `echo-v2` as `stable` and `staging` respectively:
+```
+kn service update svc --tag-revision echo-v1:stable --tag-revision echo-v2:staging
+```
+
+2. Ramp up/down revision `echo-v3` to `20%`, adjusting other traffic to accommodate:
+```
+kn service update svc --traffic echo-v3:20 --traffic echo-v2:80
+```
+
+3. Give revision `echo-v3` the tag `candidate`, without otherwise changing any traffic split:
+```
+kn service update svc --tag-revision echo-v3:candidate
+```
+
+4. Give `echo-v3` the tag `candidate`, and `2%` of traffic adjusting other traffic to go to revision `echo-v2`:
+```
+kn service update svc --tag-revision echo-v3:candidate --traffic candidate:2 --traffic echo-v2:98
+```
+
+6. Update the tag for `echo-v3` from `candidate` to `current`:
+```
+kn service update svc --untag-revision echo-v3 --tag-revision echo-v3:current
+```
+
+7. Remove the tag `current` from `echo-v3`:
+```
+kn service update svc --untag-revision current
+```
+
+8. Remove `echo-v3` from the traffic assignments entirely, adjusting `echo-v2` to fill up:
+```
+kn service update svc --traffic echo-v3:0 --traffic echo-v2:100
+```
```

Comment:
If we decide to route 100% traffic to echo-v2 like `echo-v2:100`, I think we don't need to specify other --traffic like `echo-v3:0`.

Reply:
You are right, let me update the example.

Reply:
done